presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaTableDescriptionSupplier.java

```diff
@@ -73,43 +74,32 @@
                 if (file.isFile() && file.getName().endsWith(".json")) {
                     KafkaTopicDescription table = topicDescriptionCodec.fromJson(readAllBytes(file.toPath()));
                     String schemaName = table.getSchemaName().orElse(defaultSchema);
-                    log.debug("Kafka table %s.%s: %s", schemaName, table.getTableName(), table);
-                    builder.put(new SchemaTableName(schemaName, table.getTableName()), table);
+                    log.debug("Found Kafka Table definition for %s.%s: %s", schemaName, table.getTableName(), table);
+                    tableDefintionBuilder.put(new SchemaTableName(schemaName, table.getTableName()), table);
                 }
             }
-
-            Map<SchemaTableName, KafkaTopicDescription> tableDefinitions = builder.build();
-
+            final Map<SchemaTableName, KafkaTopicDescription> tableDefinitions = tableDefintionBuilder.build();
             log.debug("Loaded Table definitions: %s", tableDefinitions.keySet());
 
-            builder = ImmutableMap.builder();
-            for (String definedTable : tableNames) {
-                SchemaTableName tableName;
-                try {
-                    tableName = parseTableName(definedTable);
-                }
-                catch (IllegalArgumentException iae) {
-                    tableName = new SchemaTableName(defaultSchema, definedTable);
-                }
-
-                if (tableDefinitions.containsKey(tableName)) {
-                    KafkaTopicDescription kafkaTable = tableDefinitions.get(tableName);
-                    log.debug("Found Table definition for %s: %s", tableName, kafkaTable);
-                    builder.put(tableName, kafkaTable);
-                }
-                else {
-                    // A dummy table definition only supports the internal columns.
-                    log.debug("Created dummy Table definition for %s", tableName);
-                    builder.put(tableName, new KafkaTopicDescription(
-                            tableName.getTableName(),
-                            Optional.ofNullable(tableName.getSchemaName()),
-                            definedTable,
-                            Optional.of(new KafkaTopicFieldGroup(DummyRowDecoder.NAME, Optional.empty(), ImmutableList.of())),
-                            Optional.of(new KafkaTopicFieldGroup(DummyRowDecoder.NAME, Optional.empty(), ImmutableList.of()))));
-                }
-            }
-
-            return builder.build();
+            Map<SchemaTableName, KafkaTopicDescription> builder = new HashMap<>();
+
+            // A dummy table definition only supports the internal columns.
+            tableNames.stream()
```

Comment:
No need to call `stream()`. You can call `forEach()` directly on the `Set`.

Code style nit: we generally prefer to use a regular for-each loop to do simple iteration and reserve `forEach` for times when it is required, such as iterating over the result of a stream transformation, or when the body can be a method reference:
```java
values.forEach(this::process);
```
In this case, it seems like a regular for-each would be simpler.